packages/dev/core/src/Meshes/Node/Blocks/geometryOptimizeBlock.ts

```diff
@@ -0,0 +1,126 @@
+import { PropertyTypeForEdition, editableInPropertyPage } from "../../../Decorators/nodeDecorator";
+import type { NodeGeometryBuildState } from "../nodeGeometryBuildState";
+import type { FloatArray } from "../../../types";
+import { VertexData } from "../../../Meshes/mesh.vertexData";
+import { Scalar } from "../../../Maths/math.scalar";
+import { Epsilon } from "../../../Maths/math.constants";
+/**
+ * Block used to extract unique positions from a geometry
+ */
+export class GeometryOptimizeBlock extends NodeGeometryBlock {
+    /**
+     * Gets or sets a boolean indicating that this block can evaluate context
+     * Build performance is improved when this value is set to false as the system will cache values instead of reevaluating everything per context change
+     */
+    @editableInPropertyPage("Evaluate context", PropertyTypeForEdition.Boolean, "ADVANCED", { notifiers: { rebuild: true } })
+    public evaluateContext = false;
+
+    /**
+     * Creates a new GeometryOptimizeBlock
+     * @param name defines the block name
+     */
+    public constructor(name: string) {
+        super(name);
+
+        this.registerInput("geometry", NodeGeometryBlockConnectionPointTypes.Geometry);
+        this.registerOutput("output", NodeGeometryBlockConnectionPointTypes.Geometry);
+    }
+
+    /**
+     * Gets the current class name
+     * @returns the class name
+     */
+    public getClassName() {
+        return "GeometryOptimizeBlock";
+    }
+
+    /**
+     * Gets the geometry component
+     */
+    public get geometry(): NodeGeometryConnectionPoint {
+        return this._inputs[0];
+    }
+
+    /**
+     * Gets the output component
+     */
+    public get output(): NodeGeometryConnectionPoint {
+        return this._outputs[0];
+    }
+
+    protected _buildBlock(state: NodeGeometryBuildState) {
+        const func = (state: NodeGeometryBuildState) => {
+            if (!this.geometry.isConnected) {
+                return null;
+            }
+            const vertexData = this.geometry.getConnectedValue(state);
+            const newPositions: FloatArray = [];
+            const newIndicesMap: { [key: number]: number } = {};
+
+            for (let index = 0; index < vertexData.positions.length; index += 3) {
+                const x = vertexData.positions[index];
+                const y = vertexData.positions[index + 1];
+                const z = vertexData.positions[index + 2];
+
+                // check if we already have it
+                let found = false;
+                for (let checkIndex = 0; checkIndex < newPositions.length; checkIndex += 3) {
+                    if (
+                        Scalar.WithinEpsilon(x, newPositions[checkIndex], Epsilon) &&
+                        Scalar.WithinEpsilon(y, newPositions[checkIndex + 1], Epsilon) &&
+                        Scalar.WithinEpsilon(z, newPositions[checkIndex + 2], Epsilon)

```

Comment:
Just a thought (can be resolved) - would it be possible to configure the epsilon (defaults to the static Epsilon value) to allow the user a more precise "unique" definition?

Reply:
What do you mean? (Sorry I'm slow)

Reply:
oh you mean on the block itself? oh that's a good idea!

Reply:
You are testing positions against already-existing positions in the array, based on an epsilon value. This epsilon is fixed, so theoretically, if the mesh's positions are very small a lot of them will be included and the mesh will be "decimated". If we can configure the epsilon we could define the precision of the compare function